### PR TITLE
fix(*): use process.env value or default for RUST_LOG RUST_BACKTRACE

### LINKED ISSUE #13931: migrate-cli: do not override RUST_LOG from the environment
We want to be able to control engines logging independently of CLI logging. Logs are not used for any business logic at present, so there is seemingly no valid reason to override RUST_LOG.

### PR DESCRIPTION
**Pull request by Jolg42**
closes https://github.com/prisma/prisma/issues/13931

### CODE REVIEW

**Review by tomhoule — APPROVED** [review]
lgtm!

File: packages/migrate/src/MigrateEngine.ts [hunk 1]
```diff
@@ -192,10 +192,14 @@ export class MigrateEngine {
...
-            ...rest,
-            SERVER_ROOT: this.projectDir,
+            // The following environment variables can be overridden by the user.
             RUST_LOG: 'info',
             RUST_BACKTRACE: '1',
+            // Take env values from process.env (willl override values set before)
+            ...processEnv,
+            // Set SERVER_ROOT
+            // TODO why is this needed?
+            SERVER_ROOT: this.projectDir,
```
Comment (Jolg42) [inline packages/migrate/src/MigrateEngine.ts:202]:
```suggestion
```

Comment (Jolg42) [reply packages/migrate/src/MigrateEngine.ts:202]: And removed ✅

[the remark this answers is not in the document]
Comment (Jolg42) [reply packages/migrate/src/MigrateEngine.ts:202]: And removed ✅

File: packages/engine-core/src/binary/BinaryEngine.ts [hunk 2]
```diff
@@ -511,6 +508,9 @@ ${chalk.dim("In case we're mistaken, please report this to us 🙏.")}`)
       ...this.env, // user-provided env vars
       ...process.env,
       ...env,
+      // use value from process.env or use default
+      RUST_BACKTRACE: process.env.RUST_BACKTRACE ?? '1',
+      RUST_LOG: process.env.RUST_LOG ?? 'info',
```
Comment (SevInf) [inline packages/engine-core/src/binary/BinaryEngine.ts:513]:
```suggestion
      RUST_BACKTRACE: '1',
      RUST_LOG: 'info',
      ...this.env, // user-provided env vars
      ...process.env,
      ...env,
```

**Pull request merged**